packages/editor-ui/src/components/ResourceLocator/ResourceLocator.vue

```diff
@@ -22,10 +22,13 @@
 					<n8n-text color="text-dark" align="center" tag="div">
 						{{ $locale.baseText('resourceLocator.mode.list.error.title') }}
 					</n8n-text>
-					<n8n-text size="small" color="text-base" v-if="hasCredential">
+					<n8n-text size="small" color="text-base" v-if="hasCredential || credentialsNotSet">
 						{{ $locale.baseText('resourceLocator.mode.list.error.description.part1') }}
-						<a @click="openCredential">{{
-							$locale.baseText('resourceLocator.mode.list.error.description.part2')
+						<a v-if="credentialsNotSet" @click="createNewCredential">{{
+							$locale.baseText('resourceLocator.mode.list.error.description.part2.noCredentials')
+						}}</a>
+						<a v-else-if="hasCredential" @click="openCredential">{{
```

Comment:
In order to not repeat template parts you could use the condition inside a common click handler, even inline
```suggestion
						<a @click="() => hasCredential ? openCredential() : createNewCredential()">{{
```

Actually this doesn't look much nicer but at least fewer lines. I leave it up to you